src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(value) {
+    if (!value.hidden && value.hasOwnProperty('hidden')) {
```

Comment:
The condition `value.hasOwnProperty('hidden')` is used to check if the `value` is a `Herbivore`. However, this approach is not reliable because any object with a `hidden` property will pass this check. Consider using `instanceof` to ensure `value` is an instance of `Herbivore`.